tests/queries/0_stateless/01847_bad_like.reference

```diff
@@ -1,15 +1,31 @@
 1
+0
```

Comment:
Why is this zero?
`SELECT '\w' LIKE '\w%';`

Reply:
because `'\w%'` should be regex `^w.*` , then that is it.

Reply:
@pufit see example in https://github.com/ClickHouse/ClickHouse/issues/46682

we want to make sure wether we enter like->regex way, or like->substr way, they behaves samely.

d in like->substr way \w is handled as w.and in like->substr way \w is handled as w.

Reply:
maybe a better demo is https://fiddle.clickhouse.com/5cd99b64-dc92-4041-802c-9fbef1589af2

```
SELECT '\awwww' LIKE '%\wwww%';
```

= 1

but

```
SELECT '\awwww' LIKE '%\ww%w%';
```

= 0


because  `'%\wwww%'` enters like->substr way, and it becomes `substring wwww`

but `'%\ww%w%'` enters regex->substr way. to make logic similar, it should become something like `.*ww.*w.*`, but actually the \ behaves differently than like->substr way

Reply:
Hi! Sorry for the long response. For compatibility reasons, we want to interpret unknown escape sequences with keeping the backslash character '\w' as \w , not w . Therefore, this test should return 1. Otherwise, it can have a significant impact on ClickHouse users.

Reply:
> Hi! Sorry for the long response. For compatibility reasons, we want to interpret unknown escape sequences with keeping the backslash character '\w' as \w , not w . Therefore, this test should return 1. Otherwise, it can have a significant impact on ClickHouse users.

Oh, then could you mind me change the like -> substring way to '\w' as \w , to make things consistence？